### PR TITLE
POST /api/articles/:article_id/comments Completed!

### PR DESCRIPTION
**Pull request by Kingsl8yer**
POST /api/articles/:article_id/comments completed and its tests!

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -180,3 +180,41 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+          .post("/api/articles/1/comments")
+          .send({username: "butter_bridge", body: "This is a test comment"})
+          .expect(201)
+          .then(({body}) => {
+              expect(body.comment_id).toBe(19);
+              expect(body.author).toBe("butter_bridge");
+              expect(body.article_id).toBe(1);
+              expect(body.body).toBe("This is a test comment");
+          });
+  });
```
Comment (B7Bprog) [inline __tests__/index.test.js:210]: Please add another test where unnecessary properties in the post object are ignored. For example when posting with this:

`{username: "butter_bridge", body: "This is a test comment", extraProperty:"extraValue"}`

File: __tests__/index.test.js [hunk 2]
```diff
@@ -180,3 +180,41 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+
+  test("status 400: responds with error message when given an invalid article_id", () => {
+      return request(app)
+          .post("/api/articles/invalid/comments")
+          .send({username: "butter_bridge", body: "This is a test comment"})
+          .expect(400)
+          .then(({body}) => {
+              expect(body.msg).toBe("Bad request");
+          });
+  });
```
Comment (B7Bprog) [inline __tests__/index.test.js:231]: Please add another test for a bad request where the object is missing required field(s), e.g. no username or body properties.

File: __tests__/index.test.js [hunk 3]
```diff
@@ -180,3 +180,41 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+
```
Comment (B7Bprog) [inline __tests__/index.test.js:232]: Please write another test where the username is valid but can not be found in the database. What status code would you give it?

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Miguel, please add the required tests (see my comments), the please re-submit this PR to us with the changes.

File: MVC/errors.js [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
-  exports.handlePsqlErrors = (err, req, res, next) => {
-    if (err.code === '22P02') {
-      res.status(400).send({ msg: 'Bad request' });
-    } else next(err);
-  };
+exports.handlePsqlErrors = (err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request" });
+  } else if (err.code === "23503") {
+    res.status(400).send({ msg: "Bad request" });
```
Comment (NRMSMITH) [inline MVC/errors.js:9]: seeing as these two have the same outcome - could you double up the first if statement?

File: __tests__/index.test.js [hunk 5]
```diff
@@ -124,97 +124,129 @@ describe("GET /api/articles", () => {
...
+      });
+  });
 
   test("status 200: responds with an empty array when given an article_id with no comments", () => {
-      return request(app)
+    return request(app)
       .get("/api/articles/2/comments")
       .expect(200)
       .then(({ body }) => {
         expect(body.msg).toBe("Comments not found");
```
Comment (NRMSMITH) [inline __tests__/index.test.js:157]: You've said it responds with an empty array in your description, but then you're checking for a message?

File: __tests__/index.test.js [hunk 6]
```diff
@@ -124,97 +124,129 @@ describe("GET /api/articles", () => {
...
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
 
+  test("status 400: responds with error message when given non-existent username", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: "invalid", body: "This is a test comment" })
+      .expect(400)
```
Comment (NRMSMITH) [inline __tests__/index.test.js:247]: This is technically just non-existent, but it _could_ exist, so therefore could be a 404 instead of a 400

**Review by NRMSMITH — APPROVED** [review]
A couple of mostly superficial edits, please merge when done - no need for another PR 😄

**Pull request merged**